### PR TITLE
Feature metadata exiftool

### PR DESCRIPTION
**Pull request by GustavoZHA**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 1]
```diff
@@ -26,11 +35,51 @@
...
+
+        ExtractMetadata extractMetadata = new ExtractMetadata(fileToExtract, fileToExport);
+        extractMetadata.extractMetadata();
+        LOGGER.info("finish");
+        return ResponseEntity.ok("Extract metadata Complete");
+    }
+
+    /**
+     * Endpoint for extract metadata
+     */
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java:64]: Missing return statement

File: src/main/java/org/fundacion/jala/converter/service/ExtractMetadata.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    public ExtractMetadata(ObjectMetadata extractMetadata) {
+        this.fileToExtract = extractMetadata.getFileToExtract();
+        if (extractMetadata.getMoreInfo()) setMoreInformation();
+        exportTypeFile = new ExportTypeFile(fileToExtract.getName(), extractMetadata.getNameExport(), extractMetadata.getTypeFileExport(), extractMetadata.getFileToExport());
+        exportFile = exportTypeFile.getNameFileCompleteToExport();
+        extractMetadata();
+    }
+
+    public ExtractMetadata(final File fileExtract, final File fileExport) {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/service/ExtractMetadata.java:37]: Missing javadoc

File: src/main/java/org/fundacion/jala/converter/service/ExtractMetadata.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+public class ExtractMetadata {
+    private final String addressExiftool = "cd thirdparty\\window\\exiftool\\12.2.2_exiftool/";
+    private String exportFile = "";
+    private File fileToExtract;
+    private String moreInformation = " ";
+    private ExportTypeFile exportTypeFile;
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public ExtractMetadata(ObjectMetadata extractMetadata) {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/service/ExtractMetadata.java:29]: Missing javadoc

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 4]
```diff
@@ -26,11 +35,51 @@
      */
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java:35]: missing return statement

File: src/main/java/org/fundacion/jala/converter/service/ObjectMetadata.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import java.io.File;
+
+public class ObjectMetadata {
+    private File fileToExtract;
+    private File fileToExport;
+    private String nameExport;
+    private TypeFileExport typeFileExport = TypeFileExport.XMP;
+    private boolean moreInfo = false;
+
+    /**
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/service/ObjectMetadata.java:21]: The description of the methods are missing in all java docs of this class ObjectMetadata

**Review by Raymundogs — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 6]
```diff
@@ -1,23 +1,32 @@
 /**
  * Copyright (c) 2021 Fundacion Jala.
- *
+ * <p>
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: this tag should be removed from copyright

File: src/main/java/org/fundacion/jala/converter/service/ObjectMetadata.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+ * <p>
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
+package org.fundacion.jala.converter.service;
+
+import org.fundacion.jala.converter.service.metadata.TypeFileExport;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/service/ObjectMetadata.java]: these spaces in the code should be removed

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 8]
```diff
@@ -26,11 +35,51 @@
...
+    @GetMapping("/metadata")
+    public ResponseEntity<String> extractMetadataDefaultName() {
+        LOGGER.info("start");
+
+        File fileToExtract = new File("C:\\Users\\ASUS\\Desktop\\AT Materias\\Prog102\\prog102-AT13-JalaConverter\\JalaConverter\\Images\\img7.jpg");
+        File fileToExport = new File("C:\\Users\\ASUS\\Desktop\\AT Materias\\Prog102\\prog102-AT13-JalaConverter\\JalaConverter\\archive");
+
+        ExtractMetadata extractMetadata = new ExtractMetadata(fileToExtract, fileToExport);
+        extractMetadata.extractMetadata();
+        LOGGER.info("finish");
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: these spaces in the code should be removed

**Review by JessickaMoya — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 9]
```diff
@@ -1,23 +1,32 @@
...
+import org.fundacion.jala.converter.service.ExtractMetadata;
 import org.fundacion.jala.converter.service.FileStorageService;
+import org.fundacion.jala.converter.service.ObjectMetadata;
+import org.fundacion.jala.converter.service.metadata.TypeFileExport;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java:22]: these spaces in the code should be removed

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java (hunk 6 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: That's right

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java (hunk 9 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java:22]: ok

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java (hunk 4 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java:35]: We do not have this agreement for endpoint

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java (hunk 8 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: ok

File: src/main/java/org/fundacion/jala/converter/service/ObjectMetadata.java (hunk 5 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/service/ObjectMetadata.java:21]: Solve

**Review by edsonarios — APPROVED** [review]

**Review by cristianfchq — APPROVED** [review]

**Review by JorgeRCV — APPROVED** [review]

**Review by danielaSantaCruz — APPROVED** [review]

**Review by JRojas999 — APPROVED** [review]

**Review by jpgonzales27 — APPROVED** [review]

**Review by Paola-Xaq — APPROVED** [review]
Everything seems ok, good job. 👍

**Review by SaulCM — APPROVED** [review]
Could use File.separator instead off "\\"

**Pull request merged**